File: 3550-smallest-index-with-digit-sum-equal-to-index/3550-smallest-index-with-digit-sum-equal-to-index.py
class Solution:
    def smallestIndex(self, nums: List[int]) -> int:
        n = len(nums)
        def f(num: int)->int:
            s = 0
            while num:
                s += (num%10)
                num//=10
            return s
        for i in range(n):
            if f(nums[i])==i:
                return i
        return -1

File: 3550-smallest-index-with-digit-sum-equal-to-index/3550-smallest-index-with-digit-sum-equal-to-index.cpp
class Solution {
public:
    bool check(int n,int ind){
        int sum = 0;
        while(n>0){
            sum += n%10;
            n = n/10;
        }
        return sum==ind;
    }
    int smallestIndex(vector<int>& nums) {
        int n = nums.size();
        for(int i=0;i<n;i++){
            if(check(nums[i],i)) return i;
        }
        return -1;
    }
};